metricbeat/module/kafka/_meta/run.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+
+wait_for_port() {
+    count=20
+    port=$1
+    while ! nc -z localhost $port && [[ $count -ne 0 ]]; do
+        count=$(( $count - 1 ))
+        [[ $count -eq 0 ]] && return 1
+        sleep 0.5
+    done
+    # just in case, one more time
+    nc -z localhost $port
+}
+
+echo "Starting ZooKeeper"
+${KAFKA_HOME}/bin/zookeeper-server-start.sh ${KAFKA_HOME}/config/zookeeper.properties &
+wait_for_port 2181
+
+echo "Starting Kafka broker"
+mkdir -p ${KAFKA_LOGS_DIR}
+${KAFKA_HOME}/bin/kafka-server-start.sh ${KAFKA_HOME}/config/server.properties \
+    --override delete.topic.enable=true --override advertised.host.name=kafka \
```

Comment:
@urso @original-brownbear The problem here with setting `avertised.host.name` to `kafka` is that this "should" work when also the kafka test is running inside an other docker container. But if the kafka container is only started locally it should be `localhost`. I tried to figure out if there is an option ot listen on both but was not successful. Do you know if this is possible.

Reply:
@ruflin I don't think you can advertise two hosts. 
I assume you want this to work with Docker on Mac too if you're talking about `localhost`? This will be tricky to without binding to `localhost` and exposing the ports on the host network.

Maybe you should simply add an environment variable and `if` to this script when setting advertised hosts.

Then for local runs you can advertise on `localhost` and for the docker network you can use the `eth0` IP (this will also work if the Docker daemon runs locally on a Linux host, but not with Docker on Mac) via:

```sh
root@48b797207577:/# HOSTIP=`ip -4 addr show scope global dev eth0 | grep inet | awk '{print \$2}' | cut -d / -f 1`
root@48b797207577:/# echo $HOSTIP
172.17.0.2
```

Reply:
@original-brownbear Thanks for the input. Yes, that was my idea. But I'm already more then happy if it works without being flaky on CI. My hope was that the new Docker image would help but the tests are still flaky and we get errors like `NoBroker available`. It could be that our system is just overloaded. Did you see similar issues on the LS side?

Reply:
@ruflin hmm no we were pretty stable with Docker on Travis. Maybe you need to wait longer  for the Broker to come up (or have a more stable way of detecting it's up)?
Did you check if the Broker maybe dies for some reason?

Reply:
The problem I have at the moment is that locally it works and on CI it doesn't (most of the times). So I assume it is one of the above problems you mentioned, unfortunately I don't know which one. 

As I would think that the health checks are good enough to wait for the broker to be available, I'm thinking the second could be the case. Do you have any idea on why the broker could die? Or ideas on how to better detect if the broker is started. Unfortunately my experience with running Kafka is pretty limited, so any help is appreciated :-)